Grafana/UI: Add basic legend to the PieChart

oscarkilhed opened the pull request:
**What this PR does / why we need it**:
This adds a basic legend to the PieChart core plugin

oscarkilhed committed:
Add basic legend to the PieChart

Clarity-89 commented on packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx:
```diff
@@ -1,8 +1,30 @@
-  return <div>Need VizLayout in grafana/ui</div>;
+export const PieChartWithLegend: FC<Props> = ({ values, width, height, ...restProps }) => {
+  const legendItems = values.map<VizLegendItem>((value) => {
+    console.log(value.title, value.color);
```
This should probably be removed.

oscarkilhed replied:
Oh, dammit. Even greped the diff for console logs to get rid of them all, but forgot to save this file after the change. Thanks.

oscarkilhed committed:
Remove console log

hugohaggmark commented on packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx:
```diff
@@ -1,8 +1,29 @@
+  });
+
+  const legendElement = (
+    <VizLegend items={legendItems} placement="right" displayMode={LegendDisplayMode.List}></VizLegend>
```
Maybe change to this:
```suggestion
    <VizLegend items={legendItems} placement="right" displayMode={LegendDisplayMode.List} />
```

hugohaggmark commented on packages/grafana-ui/src/components/PieChart/PieChartWithLegend.tsx:
```diff
@@ -1,8 +1,29 @@
+  return (
+    <VizLayout width={width} height={height} legend={legendElement}>
+      {(vizWidth: number, vizHeight: number) => (
+        <PieChart values={values} width={vizWidth} height={vizHeight} {...restProps}></PieChart>
```
Maybe change to this:
```suggestion
        <PieChart values={values} width={vizWidth} height={vizHeight} {...restProps} />
```

hugohaggmark reviewed (commented):
Looking Good, I think the ComponentWithLegend approach is something we tested with the old Graph and I think most of us agree it wasn't a very good pattern. 
Not sure this is the scope for this PR though, but you might want to look at how we compose Legend and other abilities in the new Graph as an inspiration moving forward.

Also a friendly reminder to always add Project and Milestone on your PRs. And if this is going into the v.7.4.x branch it will also need the `backport v7.4.x` label.

torkelo reviewed (approved):
Looks good!  

Think instead of exporting PieChartWithĹegend, lets make the exported PieChart component support legend. Maybe rename PieChart to PieChartSvg or something.  

The panel needs some legend options. Like placement and what values to show. But can be done in a follow-up PR

torkelo commented on packages/grafana-ui/src/components/PieChart/PieChart.tsx:
```diff
@@ -10,6 +10,8 @@ import { useTooltip, useTooltipInPortal } from '@visx/tooltip';
 import { useComponentInstanceId } from '../../utils/useComponetInstanceId';
 import { css } from 'emotion';
 
+export const DEFAULT_COLOR = 'gray';
```
think there is a FALLBACK_COLOR exported from grafana/data

oscarkilhed committed:
Remove PieChartWithLegend and refactor PieChart a bit

oscarkilhed committed:
Use FALLBACK_COLOR

oscarkilhed commented:
> Think instead of exporting PieChartWithĹegend, lets make the exported PieChart component support legend. Maybe rename PieChart to PieChartSvg or something.

Ah, missed this. I removed the PieChartWithLegend and added the legend to the PieChart. But you're probably right in that the svg should be separated from the legend

oscarkilhed committed:
Refactor PieChart

oscarkilhed committed:
Change back to [function]
